docs/guides/chip_tool_guide.md

```diff
@@ -391,6 +391,30 @@ $ ./chip-tool basic
 This section contains a general list of various CHIP Tool commands and options,
 not limited to commissioning procedure and cluster interaction.
 
+### Interactive mode versus single command mode
+
+By default, chip-tool runs in single command mode where if any single command
+does not complete within a certain timeout period, chip-tool will exit with a
+timeout error.
+
+Example of error:
+
+```
+[1650992689511] [32397:1415601] CHIP: [TOO] Run command failure: ../../../examples/chip-tool/commands/common/CHIPCommand.cpp:392: CHIP Error 0x00000032: Timeout
+```
+
+For commands such as event subscriptions that need to run for longer periods of
+time, chip-tool can be started in interactive mode first before running the
+command. In interactive mode, there will be no timeout and multiple commands can
+be issued.
```

Comment:
Why not add a timeout argument to chip-tool's Command class that propagates down to the commands instantiated on the command line? Otherwise you need modality where you can't just do the basic expected command line flows and can also be impacted by DUTs which may take a longer time to execute commands (e.g. sleepier devices or commands that take a long time to execute on the DUT).